src/server/registry/manifest.go

```diff
@@ -49,11 +52,58 @@ func getManifest(w http.ResponseWriter, req *http.Request) {
 		req.URL.RawPath = req.URL.EscapedPath()
 	}
 
-	recorder := lib.NewResponseRecorder(w)
-	proxy.ServeHTTP(recorder, req)
+	buffer := lib.NewResponseBuffer(w)
+	// whether get manifest from cache
+	fromCache := false
+	// whether need to write-back cache
+	wbCache := false
+	// check cache
+	if config.CacheEnabled() {
+		manifest, err := pkg.ManifestMgr.Get(req.Context(), art.Digest)
+		if err == nil {
+			fromCache = true
```

Comment:
did you go through this [code](https://github.com/distribution/distribution/blob/main/registry/handlers/manifests.go#L84)?